tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,543 @@
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    assert entry.state == ConfigEntryState.LOADED
+
+
+async def test_async_setup_entry_connection_Error(
+    hass, mock_azure_data_explorer_test_connection
+):
+    """Test Error when not getting proper connection."""
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    mock_azure_data_explorer_test_connection.side_effect = Exception("Message")
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    assert entry.state == ConfigEntryState.SETUP_RETRY
+
+
+async def test_async_setup_entry_KustoService_Error(
+    hass, mock_azure_data_explorer_test_connection
+):
+    """Test Error when not getting proper connection."""
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    mock_azure_data_explorer_test_connection.side_effect = KustoServiceError("Message")
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    assert entry.state == ConfigEntryState.SETUP_ERROR
+
+
+async def test_failed_test_connection_Auth_Error(
+    hass, mock_azure_data_explorer_test_connection
+):
+    """Test Error when not getting proper connection."""
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+    mock_azure_data_explorer_test_connection.side_effect = KustoAuthenticationError(
+        authentication_method="AM", exception=Exception
+    )
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    assert entry.state == ConfigEntryState.SETUP_RETRY
+
+
+async def test_AzureDataExplorer_test_connection(hass, mock_test_connection):
+    """Test the Test_Connection method when init."""
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    AzureDataExplorer(hass, entry)
+    mock_test_connection.assert_called_once()
+
+
+async def test_AzureDataExplorer_parse_event(hass):
+    # pylint: disable=protected-access
+    """Test parsing events."""
+
+    # Pass to old event
+    entry = MockConfigEntry(
+        domain=azure_data_explorer.DOMAIN,
+        data=BASE_CONFIG_FULL,
+        title="cluster",
+        options=BASIC_OPTIONS,
+    )
+
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+
+    azure_data_explorer_instance = AzureDataExplorer(hass, entry)
```

Comment:
Shouldn't need this, you can get the instance from hass.data, but really shouldn't be testing the internal workings, just the entry and exit points, so that a change in implementation doesn't require changes to the tests